File: backend/src/main/java/cliquee/backend/model/AuditLog.java
package cliquee.backend.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import java.util.UUID;
import lombok.Data;
import org.hibernate.annotations.UuidGenerator;

@Data
@Entity
@Table(name = "audit_logs")
public class AuditLog {

  @Id
  @UuidGenerator
  @Column(updatable = false, nullable = false)
  private UUID id;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "user_id", nullable = false)
  private User user;

  private String action;
  private String details;
  private LocalDateTime created_at = LocalDateTime.now();
}

File: flaskBackend/app/models/audit_log.py
from datetime import datetime
from ..extensions import db

class Audit_Log(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)
    action = db.Column(db.String(64))
    details = db.Column(db.String(255))
    created_at = db.Column(db.DateTime, default=datetime.now())


    def __repr__(self):
        return f'<Audit_Log {self.action}>'